edenai_apis/apis/leonardo/leonardo_api.py

```diff
@@ -0,0 +1,142 @@
+import base64
+import http.client
+from typing import Dict, Generator, List, Literal, Optional, Union, overload
+import requests
+
+from edenai_apis.features import ImageInterface
+from edenai_apis.features.image.generation.generation_dataclass import (
+    GenerationDataClass,
+    GeneratedImageDataClass,
+)
+from edenai_apis.features.provider.provider_interface import ProviderInterface
+from edenai_apis.loaders.loaders import load_provider, ProviderDataEnum
+from edenai_apis.utils.exception import ProviderException
+from edenai_apis.utils.types import ResponseType
+from .config import get_model_id_image
+
+class LeonardoApi(ProviderInterface, ImageInterface):
+    provider_name = "leonardo"
+
+    def __init__(self, api_keys: Dict = {}):
+        api_settings = load_provider(
+            ProviderDataEnum.KEY, provider_name=self.provider_name, api_keys=api_keys
+        )
+        self.headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+            "Authorization": f"Token {api_settings['api_key']}",
+        }
+        self.base_url = "https://cloud.leonardo.ai/api/rest/v1"
+
+    @overload
+    def __get_response(
+        self, url: str, payload: dict
+    ) -> Generator: ...
+
+    @overload
+    def __get_response(
+        self, url: str, payload: dict
+    ) -> dict: ...
+
+    def __get_response(
+        self, url: str, payload: dict
+    ) -> Union[Generator, dict]:
+        # Launch job
+        launch_job_response = requests.post(url, headers=self.headers, json=payload)

```

Comment:
We can here also add a try catch like for the previous one, to catch unexpected errors like **timeout** errors